android/src/main/java/com/reactlibrary/JudoReactNativeModule.kt

```diff
@@ -155,4 +167,29 @@ class JudoReactNativeModule internal constructor(val context: ReactApplicationCo
         val intent = configuration.toJudoActivityIntent(it)
         it.startActivityForResult(intent, JUDO_PAYMENT_WIDGET_REQUEST_CODE)
     }
+
+    private fun handleThreeDSAuthentication(promise: Promise, service: JudoApiService, receipt: Receipt) {
+        val callback = object : ThreeDSOneCompletionCallback {
+            override fun onSuccess(success: JudoPaymentResult) {
+                if (success is JudoPaymentResult.Success) {
+                    promise.resolve(getMappedResult(success.result))
+                }
+            }
+
+            override fun onFailure(error: JudoPaymentResult) {
+                if (error is JudoPaymentResult.Error) {
```

Comment:
```
if (error is JudoPaymentResult.Error || error is JudoPaymentResult.UserCancelled) {
      promise.reject(JUDO_PROMISE_REJECTION_CODE, error.error.message)
}
```
but probably we should simply:

 promise.reject(JUDO_PROMISE_REJECTION_CODE, error.error.message ?? "Unknown error occured")